src/Generators/ControllerTagGenerator.php

```diff
@@ -39,6 +42,26 @@ protected function generateControllerObjectTags()
             if ($pageClassname !== 'Page') {
                 $this->pushMixinTag($pageClassname);
             }
+        } else if ($this->isContentController($this->className)) {
+            if (empty(self::$pageClassesCache)) {
+                self::$pageClassesCache = ClassInfo::subclassesFor(Page::class);
+            }
+
+            foreach (self::$pageClassesCache as $pageClassname) {
+                if (Config::inst()->get($pageClassname, 'controller_name') == $this->className) {
```

Comment:
Instead of foreaching all the cached page classes, can you cache the `Page => Controllername` fully?
That way, instead of foreaching, you can grab the reverse of the array by Controller name, skipping the need for the foreach every time, and only have to do it once.

I minor(?) flaw here, however, is going to be pages that have the same controller, it'll be the last (or first?) found class, and all others are not included.

This, by the way, is an existing error anyway, so I don't see too much issue with it.